### PR TITLE
Address requested changes.

### PR DESCRIPTION
**Pull request by adars-kafle**
This PR has introduced the following changes:

- Add new grid component from MUI
- Decompose the card as a separate reusable component
- Add product details routes
- Add proudct card types

### CODE REVIEW

File: frontend/src/pages/Products/components/ProductGrid.tsx [hunk 1]
```diff
@@ -1,42 +1,20 @@
...
 import { Link } from "react-router-dom";
+import products from "../../../constants/products";
+import ProductCard from "./ProductCard";
 
 const ProductGrid: React.FC = () => {
     return (
         <Grid container spacing={4}>
             {products.map((product, index) => (
-                <Grid item key={index} xs={12} sm={6} md={4}>
+                <Grid key={index} size={{ xs: 2, sm: 4, md: 4 }}>
```
Comment (nirojneupane10) [inline frontend/src/pages/Products/components/ProductGrid.tsx]: Make sure that keys in lists are unique and meaningful. Currently, you are using index as the key, which is not ideal if the list can change.

File: frontend/src/pages/Products/components/ProductCard.tsx [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { Card, CardContent, CardMedia, Typography } from "@mui/material";
+import productStyles from "./styles";
+import { ProductCardProps } from "../../../types/products";
+
+const ProductCard: React.FC<ProductCardProps> = ({ product }) => {
```
Comment (nirojneupane10) [inline frontend/src/pages/Products/components/ProductCard.tsx:6]: Handle the case where product might be null or undefined

**Review by nirojneupane10 — CHANGES_REQUESTED** [review]
Please change it

File: frontend/src/pages/Products/components/ProductCard.tsx (hunk 2 above)
Comment (adars-kafle) [reply frontend/src/pages/Products/components/ProductCard.tsx:6]: done!

**Comment by adars-kafle** [comment]
> Please change it

I have added the chages!

**Pull request merged**